ios/file_picker/Sources/file_picker/FilePickerPlugin.m

```diff
@@ -528,19 +531,37 @@ -(void)picker:(PHPickerViewController *)picker didFinishPicking:(NSArray<PHPicke
     __block NSInteger completedCount = 0;
     NSInteger totalCount = results.count;
 
+    bool isImageSelection = self.type == IMAGE;
+    bool isMediaSelection = self.type == MEDIA;
     for (NSInteger index = 0; index < results.count; ++index) {
         dispatch_group_enter(_group);
         PHPickerResult * result = [results objectAtIndex:index];
         
         dispatch_async(processQueue, ^{
             @autoreleasepool {
+            if (isMediaSelection) {
+                if (![result.itemProvider hasItemConformingToTypeIdentifier:@"public.image"] &&
+                    ![result.itemProvider hasItemConformingToTypeIdentifier:@"public.movie"]) {
+                    [errors addObject:[NSString stringWithFormat:@"Item at index %ld is not an image or video", (long)index]];
+                    dispatch_group_leave(self->_group);
+                    return;
+                }
+            } else if (isImageSelection) {
                 if (![result.itemProvider hasItemConformingToTypeIdentifier:@"public.image"]) {
                     [errors addObject:[NSString stringWithFormat:@"Item at index %ld is not an image", (long)index]];
                     dispatch_group_leave(self->_group);
                     return;
                 }
+            }
 
-                [result.itemProvider loadFileRepresentationForTypeIdentifier:@"public.image" completionHandler:^(NSURL * _Nullable url, NSError * _Nullable error) {
+            NSString *typeIdentifier;
+            if ([result.itemProvider hasItemConformingToTypeIdentifier:@"public.image"]) {
+                typeIdentifier = @"public.image";
+            } else {
+                typeIdentifier = @"public.movie";
```

Comment:
As I understand it, this bug is fixed by checking the `public.movie` _alongside_ the `public.image` one? Nice catch!

Reply:
Realistically I went through some of the prior attempts and by trial and error figured these two cases that cover image, video and mixed. The obj-c syntax is quite cryptic to me.